clients/src/main/java/org/apache/kafka/common/TopicIdPartition.java

```diff
@@ -78,6 +78,22 @@ public TopicPartition topicPartition() {
         return topicPartition;
     }
 
+    /**
+     * Checking if TopicIdPartition meant to be the same reference to same this object but doesn't have all the data.
+     * If topic name is empty and topic id is persisted then the method will relay on topic id only
+     * otherwise the method will relay on topic name.
```

Comment:
Nit: I believe this is meant to be _rely_:

```suggestion
     * If topic name is empty and topic id is persisted then the method will rely on topic id only
     * otherwise the method will rely on topic name.
```